comp/core/workloadmeta/collectors/internal/cloudfoundry/vm/cf_vm_test.go

```diff
@@ -276,25 +280,22 @@ func TestPullActiveContainer(t *testing.T) {
 	}
 
 	err := c.Pull(context.TODO())
-	assert.NoError(t, err)
-
-	evs := workloadmetaStore.GetNotifiedEvents()
-	assert.NotEmpty(t, evs)
-
-	event0 := evs[0]
+	require.NoError(t, err)
 
-	assert.Equal(t, event0.Type, workloadmeta.EventTypeSet)
-	assert.Equal(t, event0.Source, workloadmeta.SourceClusterOrchestrator)
+	require.Eventually(t, func() bool {
+		_, err := workloadmetaStore.GetContainer(activeContainerWithoutProperties.Handle())
+		return err == nil
+	}, waitFor, tick)
 
-	containerEntity, ok := event0.Entity.(*workloadmeta.Container)
-	assert.True(t, ok)
+	container, err := workloadmetaStore.GetContainer(activeContainerWithoutProperties.Handle())
+	require.NoError(t, err)
```

Comment:
nit:

Do we really need to do this assertion given that it is already preceded by an `eventually` block that does the same assertion?

If we are repeating it just to be able to get the container and do the subsequent assertions on the container, I think we can move the assertions below inside the `eventually` block above.

Reply:
No strong preference, but I think it might be a bit clearer this way because we only need to wait until the container is in workloadmeta.

The rest of fields should be correct once the container is there, so I think putting those checks outside the Eventually function is closer to what we need to test.